src/DynamoPackages/PackageDirectoryBuilder.cs

```diff
@@ -92,6 +93,18 @@ private void RemoveDyfFiles(IEnumerable<string> filePaths, IDirectoryInfo dyfDir
             }
         }
 
+        private void RemoveOtherFiles(IEnumerable<string> filePaths, IDirectoryInfo dir)
+        {
+            // Remove all files which are not listed in the files list
+            Directory.GetFiles(dir.FullName, "*", SearchOption.AllDirectories)
+                .Where(x => !x.EndsWith("pkg.json") && !filePaths.Contains(x)).ToList()
+                .ForEach(x => fileSystem.DeleteFile(x));
+
+            // Remove all backup folders
+            Directory.GetDirectories(dir.FullName, Configuration.Configurations.BackupFolderName, SearchOption.AllDirectories)
+                .ToList().ForEach(x => Directory.Delete(x));
```

Comment:
`ToList().ForEach(...`

is an antipattern. It allocates an unnecessary list. Just use `foreach`. :)